lib/dialects/abstract/query-generator.js

```diff
@@ -1520,6 +1522,195 @@ module.exports = (function() {
+        "!..": "$notBetween",
+        "notbetween": "$notBetween",
+        "nbetween": "$notBetween"
+      };
+
+      key = aliasMap[key] || key;
+      if (_.isPlainObject(value)) {
+        _.forIn(value, function (item, key) {
+          if (aliasMap[key]) {
+            value[aliasMap[key]] = item;
+            delete value[key];
+          }
+        });
+      }
+
+      if (key === undefined) {
+        if (value instanceof Utils.or || value instanceof Utils.and) {
+          key = value instanceof Utils.or ? "$or" : "$and";
+          value = value.args;
+        }
+      }
+
+      if (value instanceof Utils.literal) {
+        return this.handleSequelizeMethod(value);
+      }
+
+      // OR/AND grouping logic
+      if (key === '$or' || key === '$and') {
+        binding = key === '$or' ? 'OR' : 'AND';
+
+        if (Array.isArray(value)) {
+          return '('+value.map(function (item) {
+            return self.whereItemsQuery(item, options, binding);
+          }).join(' '+binding+' ')+')';
+        } else {
+          return '('+self.whereItemsQuery(value, options, binding)+')';
+        }
+      }
+
+      if (value) {
+        if (value.$or || value.$and) {
+         return '('+value.$or.map(function (_value) {
+          return self.whereItemQuery(key, _value, options);
+         }).join(value.$or ? ' OR ' : ' AND ')+')';
+        }
+      }
+
+      // If multiple keys we combine the different logic conditions
+      if (_.isPlainObject(value) && Object.keys(value).length > 1) {
+        return (function () {
+          var $items = [];
+          _.forIn(value, function (item, logic) {
+            var $where = {};
+            $where[logic] = item;
+            $items.push(self.whereItemQuery(key, $where, options));
+          });
+
+          return '('+$items.join(' AND ')+')';
+        })();
+      }
+
+      // Setup keys and comparators
+      if (options.model && options.model.rawAttributes && options.model.rawAttributes[key] && typeof options.model.rawAttributes[key].type === "string" && options.model.rawAttributes[key].type.substr(-2) === '[]') {
+        // Stupid horrible ARRAY checking, need a new way
+        value = this.escape(value, options.model.rawAttributes[key]);
+      } else if (value && (value.$in || Array.isArray(value) || (value.$not && Array.isArray(value.$not)) || value.$notIn)) {
+        comparator = 'IN';
+        if (value.$not || value.$notIn) comparator = 'NOT IN';
+        value = '('+(value.$in || value.$not || value.$notIn || value).map(function (item) {
+          return self.escape(item);
+        }).join(', ')+')'; 
+      } else if (value && (value.$between || value.$notBetween)) {
+        comparator = 'BETWEEN';
+        if (value.$notBetween) comparator = 'NOT BETWEEN';
+
+        value = (value.$between || value.$notBetween).map(function (item) {
+          return self.escape(item);
+        }).join(' AND '); 
+      } else {
+        if (_.isPlainObject(value)) {
```

Comment:
seems like this method is used at least twice, maybe a candidate for pulling out into its own method during a refactor